From 76924437bb2ca09f82b139c533519e515d40519c Mon Sep 17 00:00:00 2001
From: Herbert Xu <herbert@gondor.apana.org.au>
Date: Sat, 3 May 2008 14:17:40 +0800
Subject: [PATCH] [CD] Test __GLIBC__ instead of _GNU_SOURCE

AC_GNU_SOURCE always defines _GNU_SOURCE so testing it is pointless.
This patch changes it to test __GLIBC__ instead.

Signed-off-by: Herbert Xu <herbert@gondor.apana.org.au>
---
 ChangeLog |    1 +
 src/cd.c  |    2 +-
 2 files changed, 2 insertions(+), 1 deletions(-)

diff --git a/ChangeLog b/ChangeLog
index 1126b04..94b37c4 100644
--- a/ChangeLog
+++ b/ChangeLog
@@ -1,6 +1,7 @@
 2008-05-03  Herbert Xu <herbert@gondor.apana.org.au>
 
 	* Fixed _PATH_BSHELL warning.
+	* Test __GLIBC__ instead of _GNU_SOURCE.
 
 2008-05-02  Herbert Xu <herbert@gondor.apana.org.au>
 
diff --git a/src/cd.c b/src/cd.c
index cb6e24d..73ccde0 100644
--- a/src/cd.c
+++ b/src/cd.c
@@ -249,7 +249,7 @@ inline
 STATIC char *
 getpwd()
 {
-#ifdef _GNU_SOURCE
+#ifdef __GLIBC__
 	char *dir = getcwd(0, 0);
 	return dir ? dir : nullstr;
 #else
-- 
1.5.6.3

